examples/compute-boids/main.rs

```diff
@@ -0,0 +1,391 @@
+                ]
+            },
+            wgpu::VertexBufferDescriptor {
+                stride: 2 * 4,
+                step_mode: wgpu::InputStepMode::Vertex,
+                attributes: &[
+                    // vertex positions
+                    wgpu::VertexAttributeDescriptor {
+                        offset: 0,
+                        format: wgpu::VertexFormat::Float2,
+                        shader_location: 2,
+                    },
+                ]
+            },
+        ],
+        sample_count: 1,
+        sample_mask: !0,
+        alpha_to_coverage_enabled: false,
+    });
+
+
+    // create compute pipeline
+
+    let compute_pipeline = device.create_compute_pipeline(&wgpu::ComputePipelineDescriptor {
+        layout: &compute_pipeline_layout,
+        compute_stage: wgpu::ProgrammableStageDescriptor {
+            module: &boids_module,
+            entry_point: "main",
+        },
+    });
+
+
+    // create swap chain
+
+    let mut swap_chain = device.create_swap_chain(
+        &surface,
+        &wgpu::SwapChainDescriptor {
+            usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+            format: wgpu::TextureFormat::Bgra8UnormSrgb,
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            present_mode: wgpu::PresentMode::Vsync,
+        },
+    );
+
+
+    // create depth texture attachment
+
+    let depth_texture = device.create_texture(&wgpu::TextureDescriptor {
+        size: wgpu::Extent3d { 
+            width: size.width.round() as u32,
+            height: size.height.round() as u32,
+            depth: 1,
+        },
+        mip_level_count: 1,
+        array_layer_count: 1,
+        sample_count: 1,
+        dimension: wgpu::TextureDimension::D2,
+        format: wgpu::TextureFormat::Depth24PlusStencil8,
+        usage: wgpu::TextureUsage::OUTPUT_ATTACHMENT,
+    });
+
+
+    // buffer for the three 2d triangle vertices of each instance
+
+    let vertex_buffer_data: Vec<f32> = [-0.01, -0.02, 0.01, -0.02, 0.00, 0.02].to_vec();
+    let vertices_buffer = device.create_buffer_with_data(vertex_buffer_data.as_bytes(), 
+        wgpu::BufferUsage::VERTEX | wgpu::BufferUsage::COPY_DST);
+
+
+    // buffer for simulation parameters uniform
+
+    let sim_param_data: Vec<f32> = [
+        0.04,  // deltaT
+        0.1,   // rule1Distance
+        0.025, // rule2Distance
+        0.025, // rule3Distance
+        0.02,  // rule1Scale
+        0.05,  // rule2Scale
+        0.005  // rule3Scale
+    ].to_vec();
+    let sim_param_buffer = device.create_buffer_with_data(sim_param_data.as_bytes(), 
+        wgpu::BufferUsage::UNIFORM | wgpu::BufferUsage::COPY_DST);
+
+
+    // buffer for all particles data of type [(posx,posy,velx,vely),...]
+
+    let mut initial_particle_data: Vec<f32> = vec![0.0; (4 * NUM_PARTICLES) as usize];
+    for i in 0..NUM_PARTICLES {
+        initial_particle_data[(4 * i + 0) as usize] = 2.0 * (rand::random::<f32>() - 0.5);
```

Comment:
This could probably be written much shorter, something like a [f32; 4] being repeated NUM_PARTICLES times.
Something more concrete and not as extreme, we could do:
```rust
for chunk in initial_particle_data.chunks_mut(4) {
  chunk[0] = ...;
  chunk[1] = ...;
  chunk[2] = ...;
  chunk[3] = ...;
}
```